### DISCUSSION TITLE
async with tinytuya?

### DISCUSSION THREAD

**Discussion by InterStella0** [jasonacox/tinytuya#181] [General]
Hello, are there any plans regarding adding support for async/await methods? It's a powerful syntax that are designed specifically for io bounds such as this library. Right now I'm just putting them in an executor to avoid implementing async myself as there is no async alternative to this library but it would be convenient when this library starts to support them.

**Comment by jasonacox** [comment]
@InterStella0 Thanks for opening this discussion thread!  

The charter for TinyTuya is to be a simple to use python API library for Tuya devices, accessible to beginning python learners and advanced experts alike.  It started as a drop in replacement for [pytuya](https://github.com/clach04/python-tuya), which provided support for projects using Python 2.7 as well as 3.x.  Moving to async would mean ending 2.7 support. However, we could implement a separate (optional) module that handles the async functionality.  

I love to hear perspective on this and any volunteers to help. 😁

**Reply by InterStella0** [reply]
I see, I suppose it makes more sense to make a separate module for async. Perhaps you could open an issue and set a help wanted tag for volunteers on it?

**Reply by jasonacox** [reply]
Yes, that is a great point.  @ClusterM  already raised this as an issue here: https://github.com/jasonacox/tinytuya/issues/156.

**Reply by InterStella0** [reply]
Alright
